src/coffea/dataset_tools/preprocess.py

```diff
@@ -61,6 +68,16 @@ def _get_steps(
         tree = the_file[arg.object_path]
         num_entries = tree.num_entries
 
+        form_json = None
+        form_hash = None
+        if calculate_form:
+            form_bytes = (
+                uproot.dask(tree, ak_add_doc=True).layout.form.to_json().encode("utf-8")
+            )
+
+            form_hash = hashlib.blake2b(form_bytes).hexdigest()[:16]
```

Comment:
Why blake2b and only the first 8 bytes?

Reply:
because blake2b is fast for large input data and the first 8 bytes are random enough to ensure uniqueness (typically), just trying to save some network bandwidth and keep things reasonably appetizing to human eyes.